File: stack&queue/leetcode_844/solution.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        stack_s = []
        stack_t = []

        for char in s:
            if char == '#':
                if stack_s:
                    stack_s.pop()
            else:
                stack_s.append(char)

        for char in t:
            if char == '#':
                if stack_t:
                    stack_t.pop()
            else:
                stack_t.append(char)
        
        stringify_s = ''.join(stack_s)
        stringify_t = ''.join(stack_t)
        
        if stringify_s == stringify_t: return True
        
        return False

File: stack&queue/leetcode_844/solution.js
/**
 * @param {string} s
 * @param {string} t
 * @return {boolean}
 */

var backspaceCompare = function(s, t) {
    const stackS = [];
    const stackT = [];

    for(const char of s) {
        if(char === '#') {
            stackS.pop();
        } else {
            stackS.push(char)
        }
    }

    for(const char of t) {
        if(char === '#') {
            stackT.pop();
        } else {
            stackT.push(char)
        }
    }
    
    const stringifyS = stackS.join('');
    const stringifyT = stackT.join('');

    if(stringifyS === stringifyT) return true
    
    return false
};